manifests/resolved.pp

```diff
@@ -71,14 +71,25 @@
     enable => $_enable_resolved,
   }
 
-  $_resolv_conf_target = $use_stub_resolver ? {
-    true    => '/run/systemd/resolve/stub-resolv.conf',
-    default => '/run/systemd/resolve/resolv.conf',
-  }
-  file { '/etc/resolv.conf':
-    ensure  => 'symlink',
-    target  => $_resolv_conf_target,
-    require => Service['systemd-resolved'],
+  if $ensure == 'running' {
+    $_resolv_conf_target = $use_stub_resolver ? {
+      true    => '/run/systemd/resolve/stub-resolv.conf',
+      default => '/run/systemd/resolve/resolv.conf',
+    }
+    file { '/etc/resolv.conf':
+      ensure  => 'symlink',
+      target  => $_resolv_conf_target,
+      require => Service['systemd-resolved'],
+    }
+  } else {
+    # If systemd not running make a last ditch attempt to restore
+    # /etc/resolv.conf to something that might actually work on
+    # reboot.
+    exec { 'restore_resolv.conf_if_possible':
+      command => 'cp -f /run/systemd/resolve/resolv.conf /etc/resolv.conf',
+      onlyif  => 'test $(readlink /etc/resolv.conf) = "/run/systemd/resolve/resolv.conf"  || test $(readlink /etc/resolv.conf) = "/run/systemd/resolve/stub-resolv.conf"',
```

Comment:
I think this is actually rather elegant in that it at least gives a decent chance of working. Implementation wise I do think you should first test it's really a symlink. `readlink` on a regular file is empty so it becomes `test  = "/run/systemd/resolve/resolv.conf"`. Or quote the `$()`.

Reply:
Sure.
Have yet to actually test in real life and indeed above is very broken, it cp's to the synlink destination :-)